pkg/cloudprovider/providers/vsphere/vclib/diskmanagers/vdm.go

```diff
@@ -0,0 +1,72 @@
+package diskmanagers
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/golang/glog"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib"
+)
+
+type virtualDiskManager struct {
+	diskPath      string
+	volumeOptions vclib.VolumeOptions
```

Comment:
since virtualDiskManager and vmDiskManager are private and just implementing VirtualDiskProvider, so kept it private.

Data can be accessed using 
```
VirtualDisk.DiskPath
VirtualDisk.VolumeOptions
VirtualDisk.VMOptions 
```